tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ThreadContextTest.java

```diff
@@ -580,4 +582,161 @@ public void thirdPartyContextProvidersAreIncludedInThreadContext() {
+        try {
+            // Set non-default values
+            int newPriority = originalPriority == 4 ? 3 : 4;
+            Buffer.get().append("contextControls-test-buffer-A");
+            Label.set("contextControls-test-label-A");
+
+            //Future<Integer> future1 = unmanagedThreads.submit(
+            		
+            Callable<Integer> callable = allContext.contextualCallable(() -> {
+            	Assert.assertEquals(Buffer.get().toString(), "contextControls-test-buffer-A-B",
+            			"Context type was not propagated to contextual action.");
+
+            	Buffer.get().append("-C");
+
+            	Assert.assertEquals(Label.get(), "",
+            			"Context type that is configured to be cleared was not cleared.");
+
+            	Label.set("contextControls-test-label-C");
+            	
+            	return Thread.currentThread().getPriority();
+            });
+            
+            Buffer.get().append("-B");
+            Label.set("contextControls-test-label-B");
+
+            Future<Integer> future = unmanagedThreads.submit(() -> {
+            		Buffer.get().append("unpropagated-buffer");
+            		Label.set("unpropagated-label");
+            		Thread.currentThread().setPriority(newPriority);
+            		return callable.call();
+            });
+
+            Assert.assertEquals(future.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), Integer.valueOf(newPriority),
+                    "Callable returned incorrect value.");
+            
+            Assert.assertEquals(Buffer.get().toString(), "contextControls-test-buffer-A-B-C",
+            		"Context type was not propagated to contextual action.");
+
+            Assert.assertEquals(Label.get(), "contextControls-test-label-B",
+            		"Context type was not propagated to contextual action.");
+
+            Assert.assertEquals(Thread.currentThread().getPriority(), originalPriority,
+            		"Context type was not left unchanged on contextual action.");
+        }
+        finally {
+        	// Restore original values
+        	Buffer.set(null);
+            Label.set(null);
+            Thread.currentThread().setPriority(originalPriority);
+        }
+    }
+    
+    /**
+     * Verify that the MicroProfile Concurrency ThreadContext implementation's currentContextExecutor
+     * method can be used to create an Executor instance with the context that is captured from the
+     * current thread per the configuration of the ThreadContext builder, and that the context is
+     * applied to the thread where the Executor's execute method runs. This test case aligns with use 
+     * case of supplying a contextual Executor to a thread that is otherwise not context-aware.
+     */
+    @Test
+    public void currentContextExecutorRunsWithContext() throws InterruptedException, ExecutionException, TimeoutException {
+        ThreadContext priorityAndBufferContext = ThreadContext.builder()
```

Comment:
The name, priorityAndBufferContext, is inconsistent with the configuration that says it only propagates Buffer context.